### ISSUE TITLE
Rancher healthcheck doesn't work as expected

### ISSUE THREAD

**Issue by rayout** [containous/traefik#2279]
### Do you want to request a *feature* or report a *bug*?
bug

### What did you do?
Start traefik with racnher metadata. With enabling AND disabling healtcheack.

### What did you expect to see?

1) If healtcheck is disabled and service is updating/updated, i see service rule on the traefik web admin. And servise opend in browser.

2) if healtcheck is enabled. I want to see the same behavior at this point.

### What did you see instead?

1) If healtcheck is disabled and when service is updating i see 2 servers on service backend by 1-2 seconds. But one of this servers is no longer works. (stoped by updateding when i use "start before stoping" at racnher). At browser i see Gateway Timeout.
Perhaps this is the correct behavior?

2) if heathcheck is enabled and service updating/updated with "start before stoping" i did not see any rules of this service on traefik admin. When i confirm update in rancher, traefik show new rules 

At rancher api i see:
```
"state": "upgraded",
"healthState": "healthy",
```
But if restart traefik. I will see service rules in web admin.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0
Codename:     roquefort
Go version:   go1.9.1
Built:        2017-10-16_04:52:07PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[rancher]
domain = "dev.rz"
Watch = true
RefreshSeconds = 15
ExposedByDefault = false
EnableServiceHealthFilter = false/true (!!!)

[rancher.metadata]
IntervalPoll = false
Prefix = "/latest"
```

**Comment by rayout** [comment]
Message from debug when Heatcheck is true
```
level=debug msg="Filtering service kss/web with state of upgrading"
```

**Comment by SantoDE** [comment]
Hi @rayout ,

I'll have a look :)

**Comment by jmirc** [comment]
do you have any news? We have the same issues. Simple test is to create a whoami service "emilevauge/whoami" and to click on upgrade. You will get a 404 until you pressed the button upgraded. 
It is a big issue on our side because clients are getting 404.

**Comment by jmirc** [comment]
I think the issue is in the `rancher.go` file in the containerFilter method where the state different than running and updating-running filters the container. The `upgraded` and `upgrading` states must be included.

```
func containerFilter(name, healthState, state string) bool {
	if healthState != "" && healthState != "healthy" && healthState != "updating-healthy" {
		log.Debugf("Filtering container %s with healthState of %s", name, healthState)
		return false
	}

	if state != "" && state != "running" && state != "updating-running" {
		log.Debugf("Filtering container %s with state of %s", name, state)
		return false
	}

	return true
}
```

**Comment by guy-masse** [comment]
I have the same issue. 

Traefik loose our services when the deployment is underway (upgrading). Our client gets 404 until upgrading is completed.

**Comment by traefiker** [comment]
Closed by #2962.
